mozilla/addons-server
# Remove the chown command from entrypoint.

Fixes: mozilla/addons#14972

### Description

This PR removes the chown /deps in our docker entrypoint.sh script, instead running make update_deps. locally the latter is much faster since it caches aggressively wheras chown is rerun on every new build of the image.

Further we remove the docker-compose.deps.yml as the deps mount is reintroduced to the main docker-compose.yml file.

Finally removed the --force-recreate flag on the docker compose up command. This was intended to ensure each run of make up creates a pristine set of service containers, but it seems the performance costs might outweigh the benefits here.

### Context

Each time we run make up we (locally) build a new image and then run the containers. Since there is a new image the web/worker containers are always recreated fresh. When we ran chown on the /deps directory this meant a 1+ minute execution time on every build. It essentially cannot be cached because the files are always recreated on the fresh build.

Previously we had mounted the ./deps directory and run update_deps. This was removed as redundant work after adding the DOCKER_TARGET allowing us to specify dependencies from the build, but the trade offs do not seem to add up and the previous method is probably better long term.

If we ever could find a way to not need to modify file ownership in the containers when mounting files from the host then we could side step this issue and remove both update_deps and chown but that would likely require a total rethink of the development environment (way out of scope)

### Testing

1. empty the ./deps directory "rm -rf ./deps/**"
2. Run make up. While it is running check the logs for the olympia container. "docker compose logs olympia --follow"
3. Expect the ./deps directory to begin filling up with files.
4. After the make up command finishes the services should be running
5. Create a file in the ./deps directory. Ensure the same file exists in both the web and worker containers.
6. Run make up again. Check the olympia logs again and seee that the instal is 1) faster 2) displays logs indicating dependencies are already installed "required package is already installed" etc etc.
7. Try the same using the CI config "COMPOSE_FILE=docker-compose.yml:docker-compose.ci.yml make up". Expect the same behaviour.

### Checklist
